File: 백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int[] arr = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
		br.close();

		int gcd = gcd(arr[0], arr[1]);
		System.out.println(gcd);
		System.out.println((arr[0] * arr[1]) / gcd);
	}

	private static int gcd(int n, int m) {
		return m != 0 ? gcd(m, n % m) : n;
	}
}

File: 백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.kt
import java.util.Scanner

fun main(args : Array<String>) = with(Scanner(System.`in`)) {
    val arr = Array(2) {nextInt()}
    var ans1 = gcd(arr[0], arr[1])
    var ans2 = (arr[0] * arr[1]) / ans1
    println(ans1)
    println(ans2)
}
private fun gcd(n : Int, m : Int): Int = if (m != 0) gcd(m, n % m) else n
